internal/internal_workflow_test.go

```diff
@@ -757,6 +758,32 @@ func (s *WorkflowUnitTest) Test_CorruptedSignalOnClosedChannelWorkflow_Receive_S
 	s.EqualValues(0, len(result))
 }
 
+func bufferedChanWorkflowTest(ctx Context) error {
```

Comment:
I know this was the customer's repro but there are two things I find odd about this repro
1. Calling receive inside the for loop that is doing the sending is odd. 
2. I do not think we need the // Refill the buffer loop in order to repro this. 

Can we use a simpler repro here so its clear that the bug occurs more generally and not just in this case.

Reply:
Sure. I can create a simple one.